framework/src/base/MooseApp.C

```diff
@@ -279,6 +298,64 @@ MooseApp::MooseApp(InputParameters parameters)
 
   if (_check_input && isParamValid("recover"))
     mooseError("Cannot run --check-input with --recover. Recover files might not exist");
+
+  if (isParamValid("start_in_debugger"))
+  {
+    auto command = getParam<std::string>("start_in_debugger");
+
+    Moose::out << "Starting in debugger using: " << command << std::endl;
+
+    auto hostname = MooseUtils::hostname();
+
+    std::stringstream command_stream;
+
+    // This will start XTerm and print out some info first... then run the debugger
+    command_stream << "xterm -e \"echo 'Rank: " << processor_id() << "  Hostname: " << hostname
+                   << "  PID: " << getpid() << "'; echo ''; ";
+
+    // Figure out how to run the debugger
+    if (command.find("lldb") != std::string::npos || command.find("gdb") != std::string::npos)
+      command_stream << command << " -p " << getpid();
+    else
+      mooseError("Unknown debugger: ",
+                 command,
+                 "\nIf this is truly what you meant then contact moose-users to have a discussion "
+                 "about adding your debugger.");
+
+    // Finish up the command
+    command_stream << "\""
+                   << " & ";
+
+    std::string command_string = command_stream.str();
+    Moose::out << "Running: " << command_string << std::endl;
+
+    system(command_string.c_str());
```

Comment:
Minor: Should probably grab this from cstdlib instead to avoid namespace collision.

Reply:
👍